desmume/src/frontend/posix/shared/ctrlssdl.cpp

```diff
@@ -401,53 +589,49 @@ do_process_joystick_events( u16 *keypad, SDL_Event *event) {
             RM_KEY( *keypad, key_o );
         }
       break;
-
-    case SDL_JOYHATMOTION:
-      /* Diagonal positions will be treated as two separate keys being activated, rather than a single diagonal key. */
-      /* JOY_HAT_* are sequential integers, rather than a bitmask */
-      key_code = ((get_joystick_number_by_id(event->jhat.which) & 15) << 12) | JOY_HAT << 8 | ((event->jhat.hat & 63) << 2);
+    
+    /* Diagonal positions will be treated as two separate keys being activated, rather than a single diagonal key. */
+    /* JOY_HAT_* are sequential integers, rather than a bitmask */
+    case 1:
+      key_code = ((jn & 15) << 12) | JOY_HAT << 8 | ((bnum & 63) << 2);
       key_u = lookup_joy_key( key_code | JOY_HAT_UP );
       key_r = lookup_joy_key( key_code | JOY_HAT_RIGHT );
       key_d = lookup_joy_key( key_code | JOY_HAT_DOWN );
       key_l = lookup_joy_key( key_code | JOY_HAT_LEFT );
-      if ((key_u != 0) && ((event->jhat.value & SDL_HAT_UP) != 0))
+      if ((key_u != 0) && ((bval & SDL_HAT_UP) != 0))
         ADD_KEY( *keypad, key_u );
       else if (key_u != 0)
         RM_KEY( *keypad, key_u );
-      if ((key_r != 0) && ((event->jhat.value & SDL_HAT_RIGHT) != 0))
+      if ((key_r != 0) && ((bval & SDL_HAT_RIGHT) != 0))
         ADD_KEY( *keypad, key_r );
       else if (key_r != 0)
         RM_KEY( *keypad, key_r );
-      if ((key_d != 0) && ((event->jhat.value & SDL_HAT_DOWN) != 0))
+      if ((key_d != 0) && ((bval & SDL_HAT_DOWN) != 0))
         ADD_KEY( *keypad, key_d );
       else if (key_d != 0)
         RM_KEY( *keypad, key_d );
-      if ((key_l != 0) && ((event->jhat.value & SDL_HAT_LEFT) != 0))
+      if ((key_l != 0) && ((bval & SDL_HAT_LEFT) != 0))
         ADD_KEY( *keypad, key_l );
       else if (key_l != 0)
         RM_KEY( *keypad, key_l );
       break;
-
-      /* Joystick button pressed */
-    case SDL_JOYBUTTONDOWN:
-      key_code = ((get_joystick_number_by_id(event->jbutton.which) & 15) << 12) | JOY_BUTTON << 8 | (event->jbutton.button & 255);
+    
+    /* Joystick button pressed */
+    case 2:
+      key_code = ((jn & 15) << 12) | JOY_BUTTON << 8 | (bnum & 255);
       key = lookup_joy_key( key_code );
       if (key != 0)
         ADD_KEY( *keypad, key );
       break;
-
-      /* Joystick button released */
-    case SDL_JOYBUTTONUP:
-      key_code = ((get_joystick_number_by_id(event->jbutton.which) & 15) << 12) | JOY_BUTTON << 8 | (event->jbutton.button & 255);
+    
+    /* Joystick button released */
+    case 3:
```

Comment:
can you make this an enum ? then we dont need comments to explain magic values

Reply:
Done